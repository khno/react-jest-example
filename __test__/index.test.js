import React from 'react';
import Link from '../index';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // 手动触发回调函数
  tree.props.onMouseEnter();
  // 重渲染
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // 手动触发回调函数
  tree.props.onMouseLeave();
  // 重渲染
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});