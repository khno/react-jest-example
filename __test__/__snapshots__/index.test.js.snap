// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Delightful JavaScript Testing
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Delightful JavaScript Testing
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Delightful JavaScript Testing
</a>
`;
